plugin/lighthouse/handler.go

```diff
@@ -0,0 +1,63 @@
+package lighthouse
+
+import (
+	"context"
+	"net"
+	"strings"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/nonwriter"
+	"github.com/coredns/coredns/request"
+
+	"github.com/miekg/dns"
+)
+
+// ServeDNS implements the plugin.Handler interface.
+func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+	// Debug log that we've have seen the query. This will only be shown when the debug plugin is loaded.
+
+	a := new(dns.Msg)
+	a.SetReply(r)
+	a.Authoritative = true
+
+	services := lh.SvcsMap
+	svcName := strings.Split(state.QName(), ".")[0]
+	ip := services[svcName]
+	if ip == "" {
+		return onFailure(lh, ctx, w, r)
+	}
+	var rr dns.RR
+
+	switch state.Family() {
+	case 1:
+		rr = new(dns.A)
+		rr.(*dns.A).Hdr = dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass()}
+		rr.(*dns.A).A = net.ParseIP(ip).To4()
+	default:
+		log.Debugf("IPv6 queries not supported yet")
```

Comment:
```suggestion
		log.Debugf("Only A type queries supported")
```

Reply:
Query family is ipv4 or Ipv6, so makes better sense. A and AAAA are type of record that are returned in reply.

Reply:
I'm wondering if this should be a Warning instead...

Reply:
Way plugin model works, it is like a chain and if we can't handle, we pass on to next one. Just coz we couldn't handle, doesn't mean coreDNS will send a failure as other plugin may still respond. Also, the response msg NXDOMAIN serves as good enough indicator that no one could respond to it. Adding a warning may blow up the logs. I haven't seen *any* plugin log info/warning for these kind of errors.

Reply:
Wouldn’t `NOTIMP` be more appropriate here?

Reply:
I think this is something we may revisit later once we get better handle on how our plugin will be used. For now am erring on side of caution, i.e. assuming there might be some other plugin that might be able to handle a query we can't and that can return response. Note that `NOTIMP` response from CoreDNS doesn't mean "our" plugin doesn't implement it, but rather CoreDNS doesn't implement it. For services that are in-cluster, Kubernetes plugin comes first and it may be able to return IPv6 queries. Won't `NOTIMP` be misleading in that case?

Reply:
Based on other comments, figured out a way that I think works. Done.